bridgerelayer/substrate/client.go

```diff
@@ -0,0 +1,131 @@
+package substrate
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	gsrpc "github.com/Snowfork/go-substrate-rpc-client"
+	"github.com/Snowfork/go-substrate-rpc-client/config"
+	"github.com/Snowfork/go-substrate-rpc-client/signature"
+	"github.com/Snowfork/go-substrate-rpc-client/types"
+
+	etypes "github.com/snowfork/polkadot-ethereum/bridgerelayer/types"
+)
+
+// Client struct
+type Client struct {
+	api      *gsrpc.SubstrateAPI
+	metadata *types.Metadata
+}
+
+// NewClient foo
+func NewClient() (*Client, error) {
+
+	api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		panic(err)
+	}
+
+	client := Client{
+		api, metadata,
+	}
+
+	return &client, nil
+}
+
+// SubmitExtrinsic submits a packet
+func (client *Client) SubmitExtrinsic(appID [32]byte, packet etypes.PacketV2) (bool, error) {
```

Comment:
Let's call this 
```suggestion
func (client *Client) SubmitPacket(appID [32]byte, packet etypes.PacketV2) (bool, error) {
```
to distinguish from the general notion of an extrinsic. There might be others in the future.

Reply:
fixed in latest upload.